src/lib_ccx/ccx_encoders_smptett.c

```diff
@@ -263,11 +263,18 @@ int write_cc_buffer_as_smptett(struct eia608_screen *data, struct encoder_ctx *c
 					dbg_print(CCX_DMT_DECODER_608, "\r");
 					dbg_print(CCX_DMT_DECODER_608, "%s\n",context->subline);
 				}
+				#pragma GCC diagnostic ignored "-Wunused-variable" /* length is currently unused */
 				int length = get_decoder_line_encoded (context, context->subline, row, data);
 
+				#define char_p (char *)
```

Comment:
Uh, #defines in the middle of a C file are ugly - please remove.

Reply:
I left it there too type less code. EVERY function in there asks for explicit type conversion. Can I change unsigned types to signed here?

Reply:
I don't know. You can, but you're likely to get warnings somewhere else if you do. Give it a try I guess.

Reply:
Yeah. Unsigned to signed (and otherwise) conversions give half of the warnings.